src/libraries/System.Memory/tests/ReadOnlySpan/Split.char.cs

```diff
@@ -15,15 +15,19 @@ public static void SplitNoMatchSingleResult()
             ReadOnlySpan<char> value = "a b";
 
             string expected = value.ToString();
-            var enumerator = value.Split(',');
+            SpanSplitEnumerator<char> enumerator = value.Split(',');
+            Assert.True(enumerator.MoveNext());
+            Assert.Equal(expected, value[enumerator.Current].ToString());
+
+            enumerator = value.Split(",");
             Assert.True(enumerator.MoveNext());
             Assert.Equal(expected, value[enumerator.Current].ToString());
         }
 
         [Fact]
         public static void DefaultSpanSplitEnumeratorBehavior()
         {
-            var charSpanEnumerator = new SpanSplitEnumerator<string>();
+            var charSpanEnumerator = new SpanSplitEnumerator<char>();
```

Comment:
```suggestion
            ReadOnlySpan<char> charSpanEnumerator = new SpanSplitEnumerator<char>();
```
To align with the other changes from `var` to `ReadOnlySpan<T>`.
Same below, but for some reason cannot add a suggestion, as it is not part of the diff.